File: app/src/main/java/zebrostudio/wallr100/ui/collection/CollectionPresenterImpl.java
package zebrostudio.wallr100.ui.collection;

import javax.inject.Inject;

import zebrostudio.wallr100.data.DataManager;
import zebrostudio.wallr100.utils.FragmentTags;

/**
 * Listens to user actions from the UI ({@link CollectionFragment}), retrieves the data and updates
 * the UI as required.
 */
public class CollectionPresenterImpl implements CollectionFragmentContract.CollectionPresenter {

    private DataManager mDataManager;
    private CollectionFragmentContract.CollectionView mCollectionView;

    @Inject
    CollectionPresenterImpl(DataManager dataManager){
        mDataManager = dataManager;
    }

    @Override
    public void bindView(CollectionFragmentContract.CollectionView view) {
        mCollectionView = view;
    }

    @Override
    public void unbindView() {
        mCollectionView = null;
    }

    @Override
    public void updateCurrentFragmentTag() {
        mDataManager.setCurrentlyInflatedFragmentTag(FragmentTags.COLLECTIONS_FRAGMENT_TAG);
    }
}

File: app/src/main/java/zebrostudio/wallr100/ui/collection/CollectionPresenterImpl.kt
package zebrostudio.wallr100.ui.collection

import zebrostudio.wallr100.data.DataRepository
import javax.inject.Inject

class CollectionPresenterImpl
@Inject constructor(private var dataRepository: DataRepository) : CollectionContract.CollectionPresenter {

  private var collectionView: CollectionContract.CollectionView? = null

  override fun attachView(view: CollectionContract.CollectionView) {
    collectionView = view
  }

  override fun detachView() {
    collectionView = null
  }

  override fun updateFragmentName(name: String) {
    dataRepository.updateCurrentFragmentName(name)
  }
}